packages/flutter_tools/test/general.shard/tracing_test.dart

```diff
@@ -234,4 +234,85 @@ void main() {
       'timeToFrameworkInitMicros': 1,
     });
   });
+
+  testWithoutContext('downloadStartupTrace also downloads the timeline', () async {
+    final BufferLogger logger = BufferLogger.test();
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final FakeVmServiceHost fakeVmServiceHost = FakeVmServiceHost(requests: <FakeVmServiceRequest>[
+      ...vmServiceSetup,
+      FakeVmServiceRequest(
+        method: 'getVMTimeline',
+        jsonResponse: vm_service.Timeline(
+          timeExtentMicros: 4,
+          timeOriginMicros: 0,
+          traceEvents: <vm_service.TimelineEvent>[
+            vm_service.TimelineEvent.parse(<String, Object>{
+              'name': kFlutterEngineMainEnterEventName,
+              'ts': 0,
+            }),
+            vm_service.TimelineEvent.parse(<String, Object>{
+              'name': kFrameworkInitEventName,
+              'ts': 1,
+            }),
+            vm_service.TimelineEvent.parse(<String, Object>{
+              'name': kFirstFrameBuiltEventName,
+              'ts': 2,
+            }),
+            vm_service.TimelineEvent.parse(<String, Object>{
+              'name': kFirstFrameRasterizedEventName,
+              'ts': 3,
+            }),
+          ],
+        ).toJson(),
+      ),
+      const FakeVmServiceRequest(
+        method: 'setVMTimelineFlags',
+        args: <String, Object>{
+          'recordedStreams': <Object>[],
+        },
+      ),
+    ]);
+
+    // Validate that old tracing data is deleted.
+    final File timelineFile = fileSystem.currentDirectory.childFile('start_up_timeline.json')
+      ..writeAsStringSync('stale');
+
+    await downloadStartupTrace(fakeVmServiceHost.vmService,
+      output: fileSystem.currentDirectory,
+      logger: logger,
+    );
+
+    final dynamic kExpectedTimeline = json.decode('''
```

Comment:
if you parse the result and compare a dart map, the test will be resilient to changes in formatting.

Reply:
Done.